tm1638/tm1638.go

```diff
@@ -0,0 +1,137 @@
+/*
+TM1638 is an LED Controller
+*/
+package tm1638
+
+import (
+	"machine"
+	"time"
+)
+
+// Device wraps the pins of the TM1638.
+type Device struct {
+	/* STB - When this pin is "LO" chip accepting transmission */
+	strobe machine.Pin
+	/* CLK - DIO pin reads data at the rising edge and outputs at the falling edge */
+	clock machine.Pin
+	/* DIO - This pin outputs/inputs serial data */
+	data machine.Pin
+}
+
+const (
+	addressAutoIncrement = 0x40
+	fixedAddress         = 0x44
+	baseAddress          = 0xC0
+	maxAddress           = 0x0F
+	readKeys             = 0x42
+	brightness           = 0x88
+	maxBrightness        = 0x07
+	zeroBrightness       = 0x80
+)
+
+// Create new TM1638 device
+func New(strobe machine.Pin, clock machine.Pin, data machine.Pin) Device {
+	strobe.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	clock.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	return Device{strobe: strobe, clock: clock, data: data}
+}
+
+// Configure TM1638
+func (d *Device) Configure() {
+	d.Clear()
+	d.Brightness(maxBrightness)
+}
+
+// Clear display memory
+func (d *Device) Clear() {
+	d.sendCommand(addressAutoIncrement)
+	d.data.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	d.strobe.Low()
+	d.transmissionDelay()
+	d.write(baseAddress)
+	for i := 0; i < 16; i++ {
+		d.write(0)
+	}
+	d.strobe.High()
+}
+
+// Set display brightness
+func (d *Device) Brightness(value uint8) {
+	if value == 0 {
+		d.sendCommand(zeroBrightness)
+	} else {
+		d.sendCommand(brightness | (value & maxBrightness))
+	}
+}
+
+// Write one display memory element
+func (d *Device) Write(offset uint8, data uint8) {
```

Comment:
Is this really useful when we also have `WriteArray` (or `WriteAt`, see next comment)?

Reply:
In the driver for an IC, I want to provide an interface to all its functions described in the datasheet. The next step is to add the driver for the specific MDU1093 board. The chip driver will help those who assemble their own device. The board driver will provide convenient access to the control of specific indicators and buttons.

So I need to implement fixed address writing function.

Reply:
> In the driver for an IC, I want to provide an interface to all functions described in the datasheet.

We don't do that for most drivers, we implement the things that make sense and are needed in practice.
Remember, it's very easy to add a new method if this is needed (just make a PR!), but refactoring/maintaining code that may or may not be used at all is tedious.